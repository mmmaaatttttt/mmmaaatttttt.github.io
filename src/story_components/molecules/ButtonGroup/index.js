import React from "react";
import PropTypes from "prop-types";
import { FlexContainer, Button } from "story_components";

const ButtonGroup = ({ data }) => {
  const buttons = data.map((d, i) => (
    <Button
      key={d.hasOwnProperty("key") ? d.key : i}
      color={d.color}
      onClick={d.handleClick}
    >
      {d.buttonText}
    </Button>
  ));
  return <FlexContainer main="center">{buttons}</FlexContainer>;
};

ButtonGroup.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.any,
      color: PropTypes.string,
      buttonText: PropTypes.string.isRequired,
      handleClick: PropTypes.func.isRequired
    })
  )
};

export default ButtonGroup;
