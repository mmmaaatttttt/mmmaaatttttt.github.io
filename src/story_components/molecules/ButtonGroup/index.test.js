import React from "react";
import { render } from "@testing-library/react";
import ButtonGroup from "./";

const testData = [
  {
    key: 1,
    color: "blue",
    buttonText: "button 1",
    handleClick: () => {}
  },
  {
    key: 2,
    color: "red",
    buttonText: "another one",
    handleClick: () => {}
  }
];

it("renders without crashing", () => {
  render(<ButtonGroup data={testData} />);
});

it("matches snapshot with props and children", () => {
  const { asFragment } = render(<ButtonGroup data={testData} />);
  expect(asFragment()).toMatchSnapshot();
});
