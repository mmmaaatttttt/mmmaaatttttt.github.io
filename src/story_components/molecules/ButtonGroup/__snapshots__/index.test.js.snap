// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with props and children 1`] = `
<DocumentFragment>
  <div
    class="sc-iwsKbI QHnwj"
    width="auto"
  >
    <button
      class="sc-bxivhb pHtqT"
      color="blue"
    >
      button 1
    </button>
    <button
      class="sc-bxivhb fWaTli"
      color="red"
    >
      another one
    </button>
  </div>
</DocumentFragment>
`;
